code/parse/sexp.cpp

```diff
@@ -16372,6 +16373,48 @@ int sexp_are_wing_flags_set(int node)
 	return SEXP_TRUE; 
 }
 
+int sexp_is_ship_emp_active(int n)
+{
+	int count, num_known;
+
+	Assert(n >= 0);
+
+	count = 0;
+	num_known = 0;
+
+	while (n != -1) {
+		count++;
+
+		auto ship_entry = eval_ship(n);
+		if (!ship_entry)
+			return SEXP_FALSE;
+		if (ship_entry->status == ShipStatus::NOT_YET_PRESENT)
+			return SEXP_CANT_EVAL;
+
+		// see if the ship has already exited the mission (either through departure or destruction)
+		if (ship_entry->exited_index >= 0) {
+			return SEXP_KNOWN_FALSE;
+
+		}
+		// ship is in mission
+		else if (ship_entry->shipp) {
+			if (ship_entry->shipp->emp_intensity > 0.0f) {
+				num_known++;
```

Comment:
No need to count here.
Really, as soon as you have any ship that is _not_ counting up here, you can immediately return false as there is at least one ship missing

Reply:
Yeah you're right. I copied that from a similar sexp, but yeah, removed.